src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -217,19 +215,43 @@ public static ColumnVector fromTimestampToUtcTimestamp(ColumnVector input, ZoneI
   }
   
   public static ColumnVector fromUtcTimestampToTimestamp(ColumnVector input, ZoneId desiredTimeZone) {
-    // TODO: Remove this check when all timezones are supported
-    // (See https://github.com/NVIDIA/spark-rapids/issues/6840)
-    if (!isSupportedTimeZone(desiredTimeZone)) {
-      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
-          desiredTimeZone.toString()));
-    }
+    warnUnsupportedTimeZone(desiredTimeZone);
     cacheDatabase();
     Integer tzIndex = instance.getZoneIDMap().get(desiredTimeZone.normalized().toString());
     try (Table transitions = instance.getTransitions()) {
       return new ColumnVector(convertUTCTimestampColumnToTimeZone(input.getNativeView(),
           transitions.getNativeView(), tzIndex));
     }
   }
+
+  public static ColumnVector timeAdd(ColumnVector input, Scalar duration, ZoneId currentTimeZone) {
+    warnUnsupportedTimeZone(currentTimeZone);
+    cacheDatabase();
+    Integer tzIndex = instance.getZoneIDMap().get(currentTimeZone.normalized().toString());
+    try (Table transitions = instance.getTransitions()) {
+      return new ColumnVector(timeAddCS(input.getNativeView(), CudfAccessor.getScalarHandle(duration),
+          transitions.getNativeView(), tzIndex));
+    }
+  }
+
+  public static ColumnVector timeAdd(ColumnVector input, ColumnView duration, ZoneId currentTimeZone) {
+    warnUnsupportedTimeZone(currentTimeZone);
+    cacheDatabase();
+    Integer tzIndex = instance.getZoneIDMap().get(currentTimeZone.normalized().toString());
+    try (Table transitions = instance.getTransitions()) {
+      return new ColumnVector(timeAddCC(input.getNativeView(), duration.getNativeView(),
+          transitions.getNativeView(), tzIndex));
+    }
+  }
+
+  private static void warnUnsupportedTimeZone(ZoneId zoneId) {
```

Comment:
nit: This is not a warning, could we call it something like `assertTimeZoneSupported`?

Reply:
done.